### ISSUE TITLE
Modifying hopping_constants in a callback

### ISSUE THREAD

**Issue by RobertGregg** [SciML/JumpProcesses.jl#258]
This is in response to a discourse question:

https://discourse.julialang.org/t/jumpproblem-hopping-constants-callback/86045?u=robertgregg

Essentially, it would be nice to be able to modify `hopping_constants` in a callback to simulate a moving boundary separates two regions with different diffusion rates. It sounds like `reset_aggregated_jumps!` would have to be modified to allow this functionality.

**Comment by RobertGregg** [comment]
I just wanted to bump this issue with an update. I've been building up a [Cellular Potts Modeling package](https://robertgregg.github.io/CellularPotts.jl/dev/) and want to incorporate spatial diffusion with JumpProcesses.jl. Given that these types of models already discretize biological cells onto a grid, I think it makes a lot of sense to use a discrete model for simulating chemical reaction networks within the model.

Basically my question becomes: Does using the `hopping_constants` in a callback to match the cell's moving boundary the right approach? Or is there a better way to approach this problem? 

Sorry in advance if this isn't the best place to have this discussion. I just didn't want this issue to become an xy problem

**Comment by isaacsas** [comment]
Looks like a great package! Sorry we haven't made progress on this yet -- unfortunately this semester got busy and I haven't had time yet to get to it. I'm hoping I can make progress on it in a couple weeks once winter break starts.

**Comment by RobertGregg** [comment]
No worries! I just thought of it the other day and wanted to follow up. No rush.

The main issue is that the cells containing the chemical reactions move in these models so I have to figure out how to deal with that.
